internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -266,64 +259,11 @@ func (s *Server) registerWebhookForRepository(
 		return regResult, nil
 	}
 
-	urlUUID := uuid.New().String()
-
-	webhookUrl := fmt.Sprintf("%s/%s", url, urlUUID)
-	parsedOriginalURL, err := urlparser.Parse(webhookUrl)
+	hookUUID, githubHook, err := s.webhookManager.CreateWebhook(ctx, client, repo.Owner, repo.Name)
 	if err != nil {
-		errStr := err.Error()
-		regResult.Status.Error = &errStr
-		return regResult, nil
-	}
-
-	hook := &github.Hook{
-		Config: map[string]interface{}{
-			"url":          webhookUrl,
-			"content_type": "json",
-			"ping_url":     ping,
-			"secret":       secret,
-		},
-		Events: ghEvents,
-	}
-
-	// if we have an existing hook for same repo, delete it
-	hooks, err := client.ListHooks(ctx, repo.Owner, repo.Name)
-	if errors.Is(err, githubprovider.ErrNotFound) {
-		logger.Debug().Msg("no hooks found")
-	} else if err != nil {
+		log.Printf("error while creating webhook: %v", err)
```

Comment:
Can we use the `logger` here (and maybe put it back onto `ctx`, so this `"error while creating webhook"` message includes the repo name and org?

Reply:
Alternatively, you can point me to where in the caller the same thing is set, and use the logger from there.

Reply:
I think I got confused by the mix and match of logging styles in the control plane... I'll change it back to zerolog.

Reply:
Yeah, zerolog got put in partway through, and we haven't fully standardized.  In the future, I might want us to move back towards `slog`, but it hasn't been a priority.  Maybe a lazy August afternoon switchover.